src/brsTypes/components/RoSGNode.ts

```diff
@@ -119,6 +121,25 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         this.parent = BrsInvalid.Instance;
     }
 
+    findNodeById(node: RoSGNode, id: BrsString): RoSGNode | BrsInvalid {
+        // test current node in tree
+        let currentId = node.get(new BrsString("id"));
+        if (currentId.toString() === id.toString()) {
```

Comment:
I think we should enforce the `id` field to alway be of `BrsString` type otherwise this would fail.

Reply:
Chatted offline and decided since all `BrsTypes` have `toString` method this will not crash (but it may not do what you are expecting if you set `id = someNonStringValue`)